compiler/ballerina-lang/src/main/java/io/ballerina/projects/DependencyGraph.java

```diff
@@ -139,6 +141,27 @@ public List<T> toTopologicallySortedList() {
         return topologicallySortedNodes;
     }
 
+    public List<T> toTopologicallySortedListWithCycles() {
```

Comment:
Shall we move the logic to ` toTopologicallySortedList` and remove this?

Reply:
I did this initially before I split them into two. Since there are usages that do not consider the cycles, I kept them as two different functions. However, since the performance dip is negligible, I can merge them again.

Reply:
Since we can't get rid of the cycle detection in package resolution and also as we are returning the cached list after the first call to resolution, it won't make a difference AFAIU.